src/lib.rs

```diff
@@ -136,6 +136,22 @@ impl FrameConstants {
             frame_number: 0,
         }
     }
+    #[inline (always)]
```

Comment:
Is there a reason you added accessors for the frame number? Something like advance_frame(), or get_next_frame_constants() that returns a new FrameConstants might make more sense.